presto-hive/src/main/java/io/prestosql/plugin/hive/GenericHiveRecordCursor.java

```diff
@@ -514,11 +519,21 @@ else if (type instanceof DecimalType) {
         }
     }
 
+    private Object getPrimitiveObjectValue(int column, Object fieldData)
+    {
+        try {
+            return ((PrimitiveObjectInspector) fieldInspectors[column]).getPrimitiveJavaObject(fieldData);
+        }
+        catch (IllegalArgumentException e) {
+            throw new PrestoException(HIVE_BAD_DATA, String.format("Error parsing field value '%s' for field %s: %s", fieldData, column, e.getMessage()), e);
+        }
+    }
+
     private void validateType(int fieldId, Class<?> type)
     {
         if (!types[fieldId].getJavaType().equals(type)) {
             // we don't use Preconditions.checkArgument because it requires boxing fieldId, which affects inner loop performance
-            throw new IllegalArgumentException(format("Expected field to be %s, actual %s (field %s)", type, types[fieldId], fieldId));
+            throw new PrestoException(HIVE_BAD_DATA, String.format("Expected field to be %s, actual %s (field %s)", type, types[fieldId], fieldId));
```

Comment:
Under what circumstances can this be propagated to the user?

Reply:
Yes, this is a sanity check designed to catch programming errors. It shouldn't be possible for this to be hit, regardless of if we get a bad file. The `types` array is built in the constructor based on the column handles, so there would have to be a bug in the connector or engine to hit this.

Thus, we should leave this one alone. Now, if you've ever seen this error occur, we want to know about it, because that means there is a bug somewhere.